components/threading/channel_threads/thread_footer/thread_footer.tsx

```diff
@@ -90,18 +91,19 @@ function ThreadFooter({
 
             <FollowButton
                 isFollowing={isFollowing}
-                follow={follow}
-                unFollow={unFollow}
+                className='separated'
+                onClick={useCallback(() => {
```

Comment:
I haven't seen this pattern in hooks before -- declaring one inline. Wondering if it makes more sense to lift it to the top of the method instead for clarity, especially if the rendering logic one days unexpectedly changes to `return;` early?